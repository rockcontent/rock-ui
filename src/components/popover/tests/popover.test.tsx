import { render } from '@testing-library/react';

import * as React from 'react';
import {
  Popover,
  PopoverTrigger,
  PopoverContent,
  PopoverHeader,
  PopoverBody,
  PopoverFooter,
  PopoverArrow,
  PopoverCloseButton,
} from '../';

describe('Popover', () => {
  it('Popover renders correctly', () => {
    const { container } = render(
      <Popover placement="right-start">
        <PopoverTrigger>
          <button>Trigger</button>
        </PopoverTrigger>
        <PopoverContent>
          <PopoverArrow />
          <PopoverCloseButton />
          <PopoverHeader>Confirmation!</PopoverHeader>
          <PopoverBody>
            Are you sure you want to have that milkshake?
          </PopoverBody>
        </PopoverContent>
        <PopoverFooter>This is the footer</PopoverFooter>
      </Popover>
    );
    expect(container).toBeTruthy();
  });
});
