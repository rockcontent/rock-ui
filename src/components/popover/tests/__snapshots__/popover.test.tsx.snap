// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover Popover renders correctly 1`] = `
<DocumentFragment>
  <button
    aria-controls="popover-content-3205597192-2"
    aria-expanded="false"
    aria-haspopup="dialog"
    id="popover-trigger-3205597192-2"
  >
    Trigger
  </button>
  <div
    class="chakra-popover__popper css-0"
    data-popper-escaped=""
    data-popper-placement="right-start"
    data-popper-reference-hidden=""
    style="visibility: hidden; min-width: max-content; inset: 0 auto auto 0; margin: 0px; --popper-transform-origin: left top; position: absolute; left: 0px; top: 0px; transform: translate(8px, 0px);"
  >
    <section
      aria-describedby="popover-body-3205597192-2"
      aria-labelledby="popover-header-3205597192-2"
      class="chakra-popover__content css-1ab1jlb"
      id="popover-content-3205597192-2"
      role="dialog"
      style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.95) translateZ(0);"
      tabindex="-1"
    >
      <div
        class="chakra-popover__arrow-positioner css-0"
        data-popper-arrow=""
        style="z-index: -1; --popper-arrow-size-half: calc(var(--popper-arrow-size, 8px) / 2); --popper-arrow-offset: calc(var(--popper-arrow-size-half) * -1); position: absolute; top: 0px; transform: translate(0px, 8px);"
      >
        <div
          class="chakra-popover__arrow css-0"
          data-popper-arrow-inner=""
          style="transform: rotate(45deg); top: 0px; left: 0px; width: 100%; height: 100%; position: absolute; z-index: inherit; box-shadow: -1px 1px 1px 0 var(--popper-arrow-shadow-color);"
        />
      </div>
      <button
        aria-label="Close"
        class="css-1w3y7qm"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="chakra-icon css-onkibi"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
            fill="currentColor"
          />
        </svg>
      </button>
      <header
        class="chakra-popover__header css-0"
        id="popover-header-3205597192-2"
      >
        Confirmation!
      </header>
      <div
        class="chakra-popover__body css-0"
        id="popover-body-3205597192-2"
      >
        Are you sure you want to have that milkshake?
      </div>
    </section>
  </div>
  <footer
    class="chakra-popover__footer css-0"
  >
    This is the footer
  </footer>
</DocumentFragment>
`;
